doc/string/split.rdoc

```diff
@@ -12,47 +12,59 @@ When +field_sep+ is <tt>$;</tt>:
   the split occurs just as if +field_sep+ were given as that string
   (see below).
 
-When +field_sep+ is <tt>' '</tt> and +limit+ is +0+ (its default value),
-the split occurs at each sequence of whitespace:
+When +field_sep+ is <tt>' '</tt> (a single space), the split occurs at each sequence of whitespace.
+Any leading sequences of whitespace are removed from the string before splitting:
 
-  'abc def ghi'.split(' ')         => ["abc", "def", "ghi"]
+  'abc def ghi'.split(' ')          # => ["abc", "def", "ghi"]
   "abc \n\tdef\t\n  ghi".split(' ') # => ["abc", "def", "ghi"]
-  'abc  def   ghi'.split(' ')      => ["abc", "def", "ghi"]
-  ''.split(' ')                    => []
-
-When +field_sep+ is a string different from <tt>' '</tt>
-and +limit+ is +0+,
-the split occurs at each occurrence of +field_sep+;
-trailing empty substrings are not returned:
-
-  'abracadabra'.split('ab')  => ["", "racad", "ra"]
-  'aaabcdaaa'.split('a')     => ["", "", "", "bcd"]
-  ''.split('a')              => []
-  '3.14159'.split('1')       => ["3.", "4", "59"]
+  'abc  def   ghi'.split(' ')       # => ["abc", "def", "ghi"]
+  ''.split(' ')                     # => []
+  '  abc  def'.split(' ')           # => ["abc", "def"]
+
+When +field_sep+ is <tt>''</tt> (an empty string), the string is split into indivisual characters.
+
+  'abc'.split('')  # => ["a", "b", "c"]
+
+When +field_sep+ is a string other than <tt>' '</tt> or <tt>''</tt>,
+the split occurs at each occurrence of +field_sep+:
+
+  'abracadabra'.split('ab')   # => ["", "racad", "ra"]
+  ''.split('a')               # => []
+  '3.14159'.split('1')        # => ["3.", "4", "59"]
   '!@#$%^$&*($)_+'.split('$') # => ["!@#", "%^", "&*(", ")_+"]
-  'тест'.split('т')          => ["", "ес"]
-  'こんにちは'.split('に')     => ["こん", "ちは"]
+  'こんにちは'.split('に')    # => ["こん", "ちは"]
 
-When +field_sep+ is a Regexp and +limit+ is +0+,
-the split occurs at each occurrence of a match;
-trailing empty substrings are not returned:
+When +field_sep+ is a Regexp,
+the split occurs at each occurrence of a match:
 
   'abracadabra'.split(/ab/) # => ["", "racad", "ra"]
-  'aaabcdaaa'.split(/a/)   => ["", "", "", "bcd"]
-  'aaabcdaaa'.split(//)    => ["a", "a", "a", "b", "c", "d", "a", "a", "a"]
+  'aaabcdaaa'.split(/[bcd]+/)   # => ["aaa", "aaa"]
+  'aaabcdaaa'.split(//)     # => ["a", "a", "a", "b", "c", "d", "a", "a", "a"]
   '1 + 1 == 2'.split(/\W+/) # => ["1", "1", "2"]
 
```

Comment:
It may not be obvious to the user how to split on `' '` without special handling.  Do you think it's worth adding an example telling the user to split on `/ /` instead of `' '` if they want to split on each single space?

Reply:
Indeed, it can be somewhat confusing that splitting by a single space requires `/ /` instead of `' '`.
I believe it would be helpful to include this as an example.